src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.uap.cs

```diff
@@ -62,15 +67,60 @@ private static RemoteInvokeHandle RemoteInvoke(MethodInfo method, string[] args,
                     i++;
                 }
 
-                AppServiceResponse response = remoteExecutionService.SendMessageAsync(message).GetAwaiter().GetResult();
+                asyncOperation = remoteExecutionService.SendMessageAsync(message);
+            }
+            finally
+            {
+                if (asyncOperation == null)
+                {
+                    remoteExecutionService.Dispose();
+                }
+            }
+
+            // Wait synchronously in a background thread so as not to block the test
+            Action waitForRemoteWaitThread;
+            Thread remoteWaitThread = ThreadTestHelpers.CreateGuardedThread(out waitForRemoteWaitThread, () =>
+            {
+                using (remoteExecutionService)
+                {
+                    AppServiceResponse response = asyncOperation.GetAwaiter().GetResult();
```

Comment:
>AppServiceResponse response = asyncOperation.GetAwaiter().GetResult(); [](start = 20, length = 70)

one last comment, this is still blocking so maybe we run into some issues if any caller really doesn't want any blocking calls.